src/posit/connect/content.py

```diff
@@ -119,76 +119,108 @@ def restart(self) -> None:
+        max_processes : int, optional
+            Max concurrent processes allowed. Default is None. Falls back to server setting 'Scheduler.MaxProcesses'.
+        min_processes : int, optional
+            Min concurrent processes required. Default is None. Falls back to server setting 'Scheduler.MinProcesses'.
+        max_conns_per_process : int, optional
+            Max client connections per process. Default is None. Falls back to server setting 'Scheduler.MaxConnsPerProcess'.
+        load_factor : float, optional
+            Aggressiveness in spawning new processes (0.0 - 1.0). Default is None. Falls back to server setting 'Scheduler.LoadFactor'.
+        cpu_request : float, optional
+            Min CPU units required (1 unit = 1 core). Default is None. Falls back to server setting 'Scheduler.CPURequest'.
+        cpu_limit : float, optional
+            Max CPU units allowed. Default is None. Falls back to server setting 'Scheduler.CPULimit'.
+        memory_request : int, optional
+            Min memory (bytes) required. Default is None. Falls back to server setting 'Scheduler.MemoryRequest'.
+        memory_limit : int, optional
+            Max memory (bytes) allowed. Default is None. Falls back to server setting 'Scheduler.MemoryLimit'.
+        amd_gpu_limit : int, optional
+            Number of AMD GPUs allocated. Default is None. Falls back to server setting 'Scheduler.AMDGPULimit'.
+        nvidia_gpu_limit : int, optional
+            Number of NVIDIA GPUs allocated. Default is None. Falls back to server setting 'Scheduler.NvidiaGPULimit'.
+        run_as : str, optional
+            UNIX user to execute the content. Default is None. Falls back to server setting 'Applications.RunAs'.
+        run_as_current_user : bool, optional
+            Run process as the visiting user (for app content). Default is False.
+        default_image_name : str, optional
+            Default image for execution if not defined in the bundle. Default is None.
+        default_r_environment_management : bool, optional
+            Manage R environment for the content. Default is None.
+        default_py_environment_management : bool, optional
+            Manage Python environment for the content. Default is None.
+        service_account_name : str, optional
+            Kubernetes service account name for running content. Default is None.
+
+        Returns
+        -------
+        None
         """
         ...
 
     @overload
-    def update(self, *args, **kwargs) -> None:
-        """Update the content item."""
+    def update(self, **attributes: Any) -> None:
+        """Update the content."""
         ...
 
-    def update(self, *args, **kwargs) -> None:
-        """Update the content item."""
-        body = dict(*args, **kwargs)
-        url = self.params.url + f"v1/content/{self.guid}"
-        response = self.params.session.patch(url, json=body)
+    def update(self, **attributes: Any) -> None:
```

Comment:
ahh, I love this - definitely much easier to understand what's going on.

Reply:
🙌🏻